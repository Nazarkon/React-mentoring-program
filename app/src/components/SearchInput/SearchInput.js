import React from 'react';
import PropTypes from 'prop-types';
import './SearchInput.scss';

const SearchInput = (props) => {
  const { defaultValue, onSearch } = props;

  const handleSubmitForm = (event) => {
    event.preventDefault();
    const data = Object.fromEntries(new FormData(event.target));
    onSearch(data.search);
  };

  const element = (
    <form id="search-form" role="search" onSubmit={handleSubmitForm}>
      <label className="search-label">FIND YOUR MOViE</label>
      <div className="search-container">
        <input
          type="text"
          className="search-input"
          aria-label="search"
          name="search"
          defaultValue={defaultValue}
          placeholder="What do you want to watch?"
        />
        <button type="submit" className="search-button">
          Search
        </button>
      </div>
    </form>
  );
  return element;
};

SearchInput.propTypes = {
  defaultValue: PropTypes.string,
  onSearch: PropTypes.func.isRequired
};

export default SearchInput;
