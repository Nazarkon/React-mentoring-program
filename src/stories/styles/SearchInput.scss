.search-form{
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
}

.search-label{
    font-family: 'Montserrat';
    font-style: normal;
    font-weight: 300;
    font-size: 40px;
    line-height: 49px;
    text-align: center;
    letter-spacing: 1px;
    text-transform: uppercase;
    color: #000000;
    align-self: flex-start;
}

.search-input {
    width: 100%;
    max-width: 700px;
    min-width: 300px;
    height: 50px;
    font-size: 20px;
    background: rgba(50, 50, 50, 0.8);
    color: white;
    mix-blend-mode: normal;
    border: 0;
    border-radius: 4px;
}

input::placeholder {
    font-size: 20px;
    color: white;
}

.search-button {
    border: 0;
    padding: 16px 70px;
    margin-left: 10px;
    background: #F65261;
    color: #FFFF;
    border-radius: 4px;
    font-family: 'Montserrat';
    font-style: normal;
    font-weight: 500;
    font-size: 20px;
    line-height: 20px;
    text-align: center;
    text-transform: uppercase;
}
.search-container{
    display: flex;
    justify-content: center;
    align-items: center;
}