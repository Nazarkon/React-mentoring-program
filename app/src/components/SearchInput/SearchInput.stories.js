import React from 'react';
import SearchInput from './SearchInput';

export default {
  title: 'Components/SearchInput',
  component: SearchInput,
  argTypes: {
    defaultValue: {
      control: {
        type: 'text',
        defaultValue: '',
      },
    },
    onSearch: {
      action: 'searched',
    },
  },
};

function Template(args) {
  return <SearchInput {...args} />;
}

export const Default = Template.bind({});
Default.args = {};
