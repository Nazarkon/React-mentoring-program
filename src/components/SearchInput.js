import React, {useState} from "react"
import "./SearchInput.scss"

const SearchInput = (props) => {

    const { defaultValue , onSearch} = props;

    const [searchValue, setSearchValue] = useState(defaultValue || '')


    const element = ( 
        <form id="search-form" role="search">
        <label className="search-label">FIND YOUR MOViE</label>
            <div className="search-container">
                <input 
                  type="text"
                  className="search-input"
                  aria-label="search"
                  name="search"
                  value={searchValue}
                  onChange={event => setSearchValue(event.target.value)}
                  placeholder="What do you want to watch?"
                  />
                <button 
                    className="search-button"
                    onClick={() => onSearch(searchValue)}
                    >
                    Search
                </button>
            </div>
        </form>
    )
    return element
}

export default SearchInput;