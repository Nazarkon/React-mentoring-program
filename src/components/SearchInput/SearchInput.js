import React, {useState} from "react"
import "./SearchInput.scss"

const SearchInput = (props) => {

    const { defaultValue , onSearch} = props;

    const [searchValue, setSearchValue] = useState(defaultValue || '')

    const handleClick = (e) => {
        e.preventDefault();
        onSearch(searchValue)
    }

    

    const element = ( 
        <form id="search-form" role="search">
        <label className="search-label">FIND YOUR MOViE</label>
            <div className="search-container">
                <input 
                  type="text"
                  className="search-input"
                  aria-label="search"
                  name="search"
                  value={searchValue}
                  onChange={event => setSearchValue(event.target.value)}
                  placeholder="What do you want to watch?"
                  />
                <button 
                    className="search-button"
                    onClick={(e) => handleClick(e)}
                    >
                    Search
                </button>
            </div>
        </form>
    )
    return element
}

export default SearchInput;