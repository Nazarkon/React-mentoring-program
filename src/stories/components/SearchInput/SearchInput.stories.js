import React from "react";
import SearchInput from "./SearchInput";

export default {
  title: "Components/SearchInput",
  component: SearchInput,
  argTypes: {
    defaultValue: {
      control: {
        type: "text",
        defaultValue: "",
      },
    },
    onSearch: {
      action: "searched",
    },
  },
};

const Template = (args) => <SearchInput {...args} />;

export const Default = Template.bind({});
Default.args = {};