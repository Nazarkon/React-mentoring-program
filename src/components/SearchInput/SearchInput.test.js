import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom";
import SearchInput from "./SearchInput";
import { BrowserRouter } from "react-router-dom";

describe("Check SearchInput element behavior", () => {
  it("Matches snapshot", () => {
    const { container } = render(
      <BrowserRouter>
        <SearchInput />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });

  test("Check that component renders an input with the value equal to initial value passed in props", () => {
    const propValue = "Search something";
    const onSearchMock = jest.fn();

    render(
      <BrowserRouter>
        <SearchInput defaultValue={propValue} onSearch={onSearchMock} />
      </BrowserRouter>
    );

    const searchInput = screen.getByLabelText("search");

    expect(searchInput).toHaveValue(propValue);
  });

  test("Check that after typing to the input and a click event on the Submit button, the onChange prop is called with proper value", () => {
    const onSearchMock = jest.fn();

    render(
      <BrowserRouter>
        <SearchInput onSearch={onSearchMock} />
      </BrowserRouter>
    );

    const searchInput = screen.getByLabelText("search");

    const submitButton = screen.getByRole("button");

    userEvent.type(searchInput, "Hello World");
    userEvent.click(submitButton);

    expect(searchInput.value).toBe("Hello World");
  });

  test("Check that after typing to the input and pressing Enter key, the onChange prop is called with proper value", () => {
    const onSearchMock = jest.fn();

    render(
      <BrowserRouter>
        <SearchInput onSearch={onSearchMock} />
      </BrowserRouter>
    );

    const searchInput = screen.getByLabelText("search");

    userEvent.type(searchInput, "Hello World");

    expect(searchInput.value).toBe("Hello World");
  });
});
