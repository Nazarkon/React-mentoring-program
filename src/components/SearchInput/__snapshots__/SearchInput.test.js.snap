// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check SearchInput element behavior Matches snapshot 1`] = `
<div>
  <form
    id="search-form"
    role="search"
  >
    <label
      class="search-label"
    >
      FIND YOUR MOViE
    </label>
    <div
      class="search-container"
    >
      <input
        aria-label="search"
        class="search-input"
        name="search"
        placeholder="What do you want to watch?"
        type="text"
        value=""
      />
      <button
        class="search-button"
        type="submit"
      >
        Search
      </button>
    </div>
  </form>
</div>
`;
